packages/api/src/external/commonwell-v2/patient/patient.ts

```diff
@@ -622,6 +598,80 @@ async function updatePatient({
   return updatedId;
 }
 
+/**
+ * Runs the patient linking flow with retries.
+ *
+ * As we upgrade links, the search fans out to find more potential links.
+ */
+async function runPatientLinkingWithRetries({
+  commonWell,
+  patient,
+  commonwellPatientId,
+  context,
+  getOrgIdExcludeList,
+}: {
+  commonWell: CommonWellAPI;
+  patient: Patient;
+  commonwellPatientId: string;
+  context: string;
+  getOrgIdExcludeList: () => Promise<string[]>;
+}): Promise<{
+  validLinks: NetworkLink[];
+  invalidLinks: NetworkLink[];
+}> {
+  let validLinks: NetworkLink[] = [];
+  let invalidLinks: NetworkLink[] = [];
+  let attempt = 0;
+
+  while (attempt < MAX_ATTEMPTS_PATIENT_LINKING) {
+    attempt++;
+    const { log } = out(
+      `retryPatientLinkingFlow - attempt ${attempt}/${MAX_ATTEMPTS_PATIENT_LINKING} - patientId ${patient.id}`
+    );
+
+    const [existingLinks, probableLinks] = await Promise.all([
+      getExistingLinks({
+        commonWell,
+        commonwellPatientId,
+      }),
+      getProbableLinks({
+        commonWell,
+        commonwellPatientId,
+      }),
+    ]);
```

Comment:
From the convo today they make it seem we should issue those in sequence. Thoughts?